### DISCUSSION TITLE
Utilising diff-tree

### DISCUSSION THREAD

**Discussion by DT-Forcology** [gitpython-developers/GitPython#1359] [Q&A]
I'm trying to utilise GitPython as part of a script that needs to act differently on folders that have been modified, essentially I need to run the following command in my script and then act accordingly on the folders if they have been modified or added:

`git diff-tree lastDeployCommit:apis headCommit:apis`

Is this possible with GitPython?

**Comment by Byron** [comment]
There is a [section in the documentation](https://gitpython.readthedocs.io/en/stable/tutorial.html#obtaining-diff-information) which describes how diffing works. From there it should be straightforward make decisions based on the results.

**Reply by DT-Forcology** [reply]
Unfortunately I didn't find the documentation very helpful, I noted that there are kwargs you can enter and I am using them. However it isn't clear how to run diff-tree.

So far I have this: diff = lastDeployCommit.diff(headCommit,ignore_all_space=True), which doesn't return what I need

**Reply by Byron** [reply]
Here is the [docs for the `diff` method](https://gitpython.readthedocs.io/en/stable/reference.html?highlight=diff#git.diff.Diffable.diff) which includes how to diff against a tree. It was found using the search field in the documentation.

<img width="1074" alt="Screenshot 2021-10-13 at 07 57 01" src="https://user-images.githubusercontent.com/63622/137044442-cfb686c2-117d-47ca-95e0-569264a5659a.png">

It's not particularly easy to use, and you are welcome to submit a PR with improvements to the documentation as you would know best what would have helped when you first arrived.
